src/watcher/on-watch.js

```diff
@@ -0,0 +1,23 @@
+const { defaultWatchEvent } = require('../constants')
+
+const scriptLoader = require('./script-loader')
+
+/**
+ * Callback used within the `on` listener in chokidar. Is executed every time a file is changed under a given `filePath`.
+ * Can execute an additional custom script if a --script argument is passed to the program.
+ * @param {string} filePath - the file path to the file or directory to watch
+ * @param {string} event - the different chokidar `on` events
+ * @param {Object} app - Custom Next.js Dev Server
+ * @param {Commander} program - Commander program
+ */
+async function onWatch(filePath, event = defaultWatchEvent, app, program) {
+  app.server.hotReloader.send('building')
+
+  if (program.script) {
+    scriptLoader(filePath, event, program)
+  }
+
+  app.server.hotReloader.send('reloadPage')
```

Comment:
@jescalan  so the proposal here would be that rather than calling the API directly, we could call an abstraction that exposes a `.send` or `.message` method.

In this way, within the abstraction, we can check which API is available within the `app` instance, allowing support for at least v9 to v10+